src/renderer/components/status-bar/status-bar-items.injectable.tsx

```diff
@@ -21,13 +22,17 @@ interface Dependencies {
   registrations: IComputedValue<StatusBarRegistration[]>;
 }
 
-function getRegisteredStatusBarItems({ registrations }: Dependencies): IComputedValue<RegisteredStatusBarItems> {
+function getStatusBarItems({ registrations }: Dependencies): IComputedValue<StatusBarItems> {
   return computed(() => {
-    const res: RegisteredStatusBarItems = {
+    const res: StatusBarItems = {
       left: [],
       right: [],
     };
 
+    // add Lens specific components
+    res.left.push(AutoUpdateComponent);
```

Comment:
For the first item, sure. When next time emerges, we need to make it comply with Open Closed Principle so that we treat internal status bar items same as items belonging to extensions. 

Fix is easy though, and if you want you can do it immediately.

Reply:
@jim-docker For the examples I provided in earlier comment, notice that #5662 simplifies it even more.